cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -1076,6 +1101,37 @@ private RetryableProvisionException handleApiException(ApiException e) throws Re
     throw new DataprocRuntimeException(e);
   }
 
+  //Throws retryable Exception for the cases that are transient in nature
+  private static void handleRetryableExceptions(Exception e) throws RetryableProvisionException {
+    // if there was an SocketTimeoutException ( read time out ) , we can just try again
+    if (e instanceof SocketTimeoutException) {
+      throw new RetryableProvisionException(e);
+    }
+
+    //Attempt retry in case of rate limit errors and service unavailable
+    if (e instanceof GoogleJsonResponseException) {
+      GoogleJsonResponseException gError = ((GoogleJsonResponseException) e);
+      int statusCode = gError.getStatusCode();
+
+      if (statusCode == HttpURLConnection.HTTP_UNAVAILABLE) {
+        throw new RetryableProvisionException(e);
+      }
+
+      if (statusCode == HttpURLConnection.HTTP_FORBIDDEN || statusCode == 429) {
+        boolean isRetryAble = gError.getDetails().getErrors().stream()
+          .anyMatch(errorInfo ->
+                      errorInfo.getReason().equalsIgnoreCase("rateLimitExceeded") ||
```

Comment:
Reason is nullable in ErrorInfo. Please reverse the check to handle null ("rateLimitExceeded".equalsIgnoreCase). BTW: Do you really need to ignore case? If not, you can just have a short set of valid codes and use contains